### PR TITLE
Remove copy-paste in ZooKeeper response handling code

### PR DESCRIPTION
**Pull request by myrrc**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 974339cfb86663991a00a31bd20a59388e1ce9d0 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50092/974339cfb86663991a00a31bd20a59388e1ce9d0/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Common/ZooKeeper/ZooKeeper.cpp [hunk 1]
```diff
@@ -876,284 +876,170 @@ Int64 ZooKeeper::getClientID()
...
+template <class T>
+auto makeNothrowCallback(std::shared_ptr<std::promise<T>> promise)
 {
-    auto promise = std::make_shared<std::promise<Coordination::CreateResponse>>();
-    auto future = promise->get_future();
+    return [promise = std::move(promise)](const T& response) mutable { promise->set_value(response); };
+}
 
-    auto callback = [promise](const Coordination::CreateResponse & response) mutable
+template <Coordination::Error ...errs, class T>
```
Comment (KochetovNicolai) [inline src/Common/ZooKeeper/ZooKeeper.cpp:893]: I like some tiny copy-paste more than variadic templates for error codes.

Comment (myrrc) [reply src/Common/ZooKeeper/ZooKeeper.cpp:893]: Well, this is some sort of black box that does the job. On the contrary, I believe that this is a better solution than manually repeating similar lambdas.

**Comment by alexey-milovidov** [comment]
The code is ok as is.

**Pull request closed without merging**